src/EnergyPlus/ScheduleManager.cc

```diff
@@ -2848,6 +2848,12 @@ namespace ScheduleManager {
             if (WhichHour <= 24) {
                 Schedule(ScheduleIndex).CurrentValue = DaySchedule(DaySchedulePointer).TSValue(TimeStep, WhichHour);
             } else if (TimeStep <= NumOfTimeStepInHour) {
+                WeekSchedulePointer = Schedule(ScheduleIndex).WeekSchedulePointer(DayOfYear_Schedule + 1);
+                if (DayOfWeek <= 7 && HolidayIndex > 0) {
+                    DaySchedulePointer = WeekSchedule(WeekSchedulePointer).DaySchedulePointer(7 + HolidayIndex);
+                } else {
+                    DaySchedulePointer = WeekSchedule(WeekSchedulePointer).DaySchedulePointer(DayOfWeek);
+                }
```

Comment:
I suggest that lines 2836-2845 be moved into each of the IF blocks (I've done that for this ELSE IF to test if it works correctly). The IF blocks are executed every time through, so there is no computational savings by having these lines outside this IF block, especially if something different needs to be done based on the conditionals. And adding extra logic on line 2838 will also cost computational time (i.e., to get the right week schedule pointer). It adds a few lines of duplicate code, but would save a little on execution time.